book/src/proptest-derive/errors.md

```diff
@@ -0,0 +1,631 @@
+    V2,
+}
+```
+
+`skip` takes no configuration. The correct form is simply `#[proptest(skip)]`.
+
+## E0021
+
+This error happens if `#[proptest(weight = <integer>)]` is passed an invalid
+integer or passed nothing at all.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+enum Foo {
+    #[proptest(weight)]
+    V1,
+    #[proptest(weight = heavy)]
+    V2,
+}
+```
+
+The only acceptable form is `#[proptest(weight = <integer>)]`, where
+`<integer>` is either an integer literal which fits in a `u32` or the same but
+enclosed in quotation marks.
+
+## E0022
+
+This error occurs if both `#[proptest(no_params)]` and `#[proptest(params = "type")]`
+are applied to the same item.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+#[proptest(no_params, params = "u8")]
+struct Foo(u32);
+```
+
+One attribute or the other must be picked depending on desired effect.
+
+## E0023
+
+This error happens if an invalid `#[proptest(params = "type")]` attribute is
+applied to an item.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+#[proptest(params = "Vec<u8")] // Note missing '>'
+struct Foo(u32);
+```
+
+There are a few different ways to get this error:
+
+- Pass nothing at all. E.g., `#[proptest(params)]`.
+
+- Pass something other than a string as the value. E.g.,
+  `#[proptest(params = 42)]`.
+
+- Pass a malformed type in the string, as in the example above. Note that type
+  syntax is limited to what the `syn` crate can understand. If you encounter a
+  valid type that is rejected here, you can work around it by defining a type
+  alias (e.g., `type MyParams = TypeThatSynCantParse;`). In such a case, please
+  consider [filing an issue](https://github.com/altsysrq/proptest/issues) as
+  well.
+
+## E0024
+
+This error happens if an invalid `#[proptest ..]` attribute is applied using a
+syntax the `proptest-derive` crate is not prepared to handle.
+
+Exactly what conditions can produce this error vary by Rust version.
+
+## E0025
+
+This error happens if both `#[proptest(strategy = "expr")]` and
+`#[proptest(value = "expr")]` are applied to the same item.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+struct Foo {
+    #[proptest(value = "42", strategy = "Just(56)")]
+    bar: u32,
+}
+```
+
+Both of these modifiers completely describe how to generate the value, so they
+cannot both be applied to the same thing. One or the other must be chosen
+depending on the desired effect.
+
+## E0026
+
+This error happens if an invalid form of `#[proptest(strategy ..)]` or
+`#[proptest(value..)]` is used.
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+struct Foo {
+    #[proptest(value = "3↑↑↑↑3")] // String content is not valid Rust syntax
+    g1: u128,
+}
+```
+
+There are a few different ways to get this error:
+
+- Pass nothing at all. E.g., `#[proptest(value)]`.
+
+- Use another illegal form. E.g., `#[proptest(value("a", "b"))]`.
+
+- Pass a string expression which is not valid Rust syntax, as in the above
+  example.
```

Comment:
```suggestion
  example. As with E0023, what is valid Rust syntax is determined by the `syn` crate.
```

Reply:
I moved the discussion about workarounds to a separate section at the bottom that everything just links to.

Reply:
Saw that; love it!